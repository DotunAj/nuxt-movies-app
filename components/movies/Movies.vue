<template>
  <div class="now-showing">
    <h3>{{ title }}</h3>
    <div class="nw-movies">
      <div v-for="movie in movies" :key="movie.popularity" class="nw-movie">
        <movie-item :movie="movie" />
      </div>
    </div>
  </div>
</template>

<script>
import MovieItem from '~/components/movies/MovieItem'
export default {
  components: {
    MovieItem
  },
  props: {
    movies: {
      type: Array,
      default: () => {}
    },
    title: {
      type: String,
      default: 'Action'
    }
  }
}
</script>
