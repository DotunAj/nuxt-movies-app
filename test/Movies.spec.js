import { shallowMount } from '@vue/test-utils'
import Movies from '~/components/movies/Movies'

const props = {
  movies: [
    {
      popularity: 1,
      posterUrl: 'mybackdrop.com/image1',
      original_title: 'A new Hope',
      releaseYear: 2019,
      genre: 'Action',
      vote_average: 7.0,
      vote_count: 500
    },
    {
      popularity: 2,
      posterUrl: 'mybackdrop.com/image2',
      original_title: 'The rise of the empire',
      releaseYear: 2019,
      genre: 'Adventure',
      vote_average: 8.0,
      vote_count: 1000
    },
    {
      popularity: 3,
      posterUrl: 'mybackdrop.com/image3',
      original_title: 'I am your father',
      releaseYear: 2019,
      genre: 'Romance',
      vote_average: 6.0,
      vote_count: 1500
    }
  ],
  title: 'Action'
}

describe('CategoryItem', () => {
  test('it should render the correct content', () => {
    const wrapper = shallowMount(Movies)
    wrapper.setProps(props)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
