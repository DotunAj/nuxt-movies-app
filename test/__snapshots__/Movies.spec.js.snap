// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem it should render the correct content 1`] = `"<div class=\\"now-showing\\"><h3>Action</h3> <div class=\\"nw-movies\\"><div class=\\"nw-movie\\"><movie-item-stub movie=\\"[object Object]\\"></movie-item-stub></div><div class=\\"nw-movie\\"><movie-item-stub movie=\\"[object Object]\\"></movie-item-stub></div><div class=\\"nw-movie\\"><movie-item-stub movie=\\"[object Object]\\"></movie-item-stub></div></div></div>"`;
